src/test/java/com/github/dockerjava/core/command/AbstractSwarmDockerClientTest.java

```diff
@@ -0,0 +1,132 @@
+package com.github.dockerjava.core.command;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.exception.ConflictException;
+import com.github.dockerjava.api.exception.NotAcceptableException;
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import com.github.dockerjava.client.AbstractDockerClientTest;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientBuilder;
+import com.github.dockerjava.core.RemoteApiVersion;
+import org.testng.ITestResult;
+import org.testng.SkipException;
+
+import java.lang.reflect.Method;
+
+import static com.github.dockerjava.utils.TestUtils.getVersion;
+
+public abstract class AbstractSwarmDockerClientTest extends AbstractDockerClientTest {
+    protected DockerClient secondDockerClient;
+    private int numberOfDockersInDocker = 0;
+
+    private static final int PORT_START = 2378;
```

Comment:
The port is the one that is used to communicate with the docker API. Picking it randomly increases chances of collision and makes it harder to debug. What are the advantages of randomly picking?